app/scripts/services/libreService.js

```diff
@@ -303,118 +325,98 @@ var libreService = function(walletService, $translate) {
                                             _scope.notifier.danger(msg, 0);
                                         });
                                     }
-                                    _scope[pendingName] = false;
+                                    if (pendingName != null) _scope[pendingName] = false;
                                 }
                                 isCheckingTx = false;
                             });
                         }, receiptInterval);
                     });
                 });
             } catch (e) {
-                _scope[pendingName] = false;
+                if (pendingName != null) _scope[pendingName] = false;
                 _scope.notifier.danger("getTransactionData: " + e);
             }
         });
     }
 
-    function canOrder(_scope, transactionFunc, rates = [0, 0]) {
-        if (rates[0] == 0 && rates[1] == 0) {
-            $translate("LIBRE_errorValidatingRates").then((msg) => {
-                _scope.notifier.danger(msg);
-            });
-            return;
-        }
-        return Promise.all([
-            getContractData("getState"),
-            getContractData("tokenBalance"),
-            getContractData(rates[0] != 0 ? "buyRate" : "sellRate"),
-            getLatestBlockData()
-        ])
-        .then((values) => {
-            let 
-                state = values[0],
-                balance = values[1],
-                rate = values[2],
-                blockData = values[3];
-
-            let time = parseInt(blockData.data.timestamp, 16);
-            let _rate = rates[0] != 0 ? rates[0] : rates[1];
-            if (rate.data[0] / coeff.rateMultiplier != _rate) {
-                $translate("LIBRE_errorValidatingRates").then((msg) => {
-                    _scope.notifier.danger(msg);
-                });
-                return;
-            }
-
-            let canOrder = (+state.data[0] == statesENUM.PROCESSING_ORDERS);
-
-            if (canOrder)
-                transactionFunc();
-            else {
-                $translate("LIBRE_orderNotAllowed").then((msg) => {
-                    _scope.notifier.danger(msg);
-                });
+    function canOrder(rates = [0, 0]) {
+        return new Promise((resolve, reject) => {
+            if (rates[0] == 0 && rates[1] == 0) {
```

Comment:
What is 0 and 1 in rates? Buy\sell?

Reply:
yes. for checking buy rate use [buy, 0], for sell - [0, sell]